mautic/mautic
# Update PHPUnit and its dependencies to 5.7 - Migrate acceptance tests to Codeception

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | N
| Related user documentation PR URL | N/A
| Related developer documentation PR URL | N/A
| Issues addressed (#s or URLs) | N/A
| BC breaks? | N/A
| Deprecations? | N/A

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
The version of PHPUnit in our composer.json was horribly outdated. This PR updates it to version 5.7 (since we still support PHP 5.6 for now) and also updates its dependencies.

This PR also introduces the [Symfony Dotenv](https://symfony.com/doc/current/components/dotenv.html) component for setting environment variables. It's currently only used in the testing environment. It allows you to set environment variables via the `.env` file in the project root. Simply copy the `.env.dist` file to `.env` and update it to match your local configuration. 

[//]: # ( As applicable: )

#### Steps to test this PR:
1. Apply PR
2. `composer install`
3. Copy the `.env.dist` file in the project root to `.env` and update it for you environment. 
4. `bin/phpunit -c app/phpunit.xml.dist`
5. All the unit tests should pass, using the env vars that you set in `.env`.
